Automattic/wp-calypso
# Jetpack Manage: Plugin management 'update plugin' link - show dialog, convert to button

Related to https://github.com/Automattic/jetpack-manage/issues/116

## Proposed Changes

* This PR adds a Dialog component to the 'Update to xxx' link that shows on the plugin management page when a plugin has an update available.
* This uses the ConfirmModal component, and adds a new prop to pass the site count through (to UpdatePlugin). This seemed better than excluding the sitecount or a link to the sites with the plugin in question, and omitting that information means the modal is still sparse in information.
* The wording is similar to the dialog wording that shows if you use the bulk edit 'Update Plugin' button at the top right of the plugin management page. That uses a different component - see `/client/my-sites/plugins/hooks/use-get-dialog-text/types.d.ts` - but for flexibility and ease of use I've opted for the ConfirmModal.
* This PR also converts the link to a button, matching the other Update Plugin button in the table header.

## Testing Instructions

* Within the Jetpack Manage dashboard, visit the plugin management page: http://jetpack.cloud.localhost:3000/plugins/manage
* Find a plugin that needs updating on a Jetpack connected site that needs updating (download an older version of a plugin on a site for testing purposes if needed).
* Click on the 'Update to' button
* Notice the modal. The wording should say the name of plugin, and the number of sites needing that plugin updated. Click ok and the plugin should update.
* The same should be visible when visiting an individual sites plugin management page
* The tests should pass as well. They will pass in the PR, but can also be tested locally: `yarn run test-client client/my-sites/plugins/plugin-management-v2/test/update-plugin.test.tsx`

### Screenshots:
On the plugin management page:
<img width="855" alt="Screenshot 2024-01-03 at 09 46 40" src="https://github.com/Automattic/wp-calypso/assets/16754605/2421343e-169c-4b6f-bd4e-01c4ed2e9967">
After clicking the 'Update to' button (showing the dialog):
<img width="822" alt="Screenshot 2024-01-04 at 13 55 46" src="https://github.com/Automattic/wp-calypso/assets/16754605/0f56b45f-7adf-4a94-98d1-9bdb2ae6d06b">


## Pre-merge Checklist
